x-pack/legacy/plugins/index_management/public/app/components/mappings_editor/components/document_fields/field_parameters/dynamic_parameter.tsx

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+
+import { documentationService } from '../../../../../services/documentation';
+import { EditFieldFormRow } from '../fields/edit_field';
+
+export const DynamicParameter = () => {
+  return (
+    <EditFieldFormRow
+      title={i18n.translate('xpack.idxMgmt.mappingsEditor.dynamicParameter.fieldTitle', {
+        defaultMessage: 'Dynamically add new properies',
```

Comment:
I chatted with Gail about this a bit. This param is related to the root-level Advanced Setting. How about we align the two by changing this title to "Dynamic child mapping" and the description to "If child fields aren't defined in the mapping, add them."

![image](https://user-images.githubusercontent.com/1238659/72651874-bd4bc500-3939-11ea-880c-42238e430695.png)

![image](https://user-images.githubusercontent.com/1238659/72651881-c046b580-3939-11ea-9c0b-a587833cb1ae.png)

Reply:
The concept of "child" is a bit strange for objects. I know that internally we refer as "children" both object "properties" and field "fields" (for multi-field).
I think the common language for an object is "properties", so here I would suggest `Dynamic property mappings`. WDYT?

Reply:
As you can see from the docs, it is about "new properties".

<img width="962" alt="Screen Shot 2020-01-20 at 13 43 18" src="https://user-images.githubusercontent.com/2854616/72709369-db5c3400-3b8a-11ea-9f3d-a01060e31391.png">

Reply:
You raise a good point. I've noticed this too, in the more general form of "We're creating a schism between our UI's terms and the API's terms" (https://github.com/elastic/kibana/issues/53546). It's a tough problem.

My approach is to construct an ideal mental model for the user with the terms we use in the UI,  using the fewest number of terms possible. Here's what I've come up with so far (terms in **bold**):

* An index's mappings is a tree of **fields**.
* A field which contains other fields is a **parent** of those fields. Those fields are the **children** of the parent field.
* A **multi-field** is a special type of field that is more conceptually similar to an **alias** than a child. It's like an alias, but located at the same place in the tree as the field its aliasing, and is of a different **data type**.

I think the important distinction between this and what's in our API docs is that this mental model is about communicating a concept, and the API docs are more about a documenting an interface. Generally speaking, I hope/believe that if the concept is clear to users, they'll be able to map it onto the API docs.  For example, if we plant the idea of "parents and children" in users' minds, and then they read the docs and see something like this:

```
PUT my_index
{
  "mappings": {
    "properties": { 
      "region": {
        "type": "keyword"
      },
      "manager": { 
        "properties": {
          "age":  { "type": "integer" },
          "name": { 
            "properties": {
              "first": { "type": "text" },
              "last":  { "type": "text" }
            }
          }
        }
      }
    }
  }
}
```

I'm hoping they'll be able to spot some patterns here and realize that this `properties` thing is the way the API defines the parent-child relationship.

So that's my list of assumptions and hypotheses. If you think this is reasonable enough, then I'd like to propose we move ahead with using our help text to convey concepts/mental models, and then carry over the terms we use in our UI over to our API docs. WDYT?

Reply:
I understand where you're coming from and in that line of thought I agree with the assumption. I personally (as a dev) think that where we might have got it wrong is in this sentence:

> An index's mappings is a tree of fields.

In my (dev) head, this concept of **tree** of fields never really "clicked". And I think that the target user to create mappings in the UI will be devs.

My personal mental model is that we have a **list** of fields and some happen to be objects.
So, if I was going to create the mappings of this model

```
const User = {
  name: 'John',
  lastName: 'Snow',
  address: {
    street: 'king_in_the_north',
    city: 'Winterfell'
  }
}
```

I would be thinking: "I need to create a mappings with **3** fields. First I will create 2 `text` type fields and then one `object` type field. To the object, I will add 2 **properties**".

I would never think that I need to create **5** fields and think about parent/child relationships. An object has properties, and obviously implicitly create that relationship for me, but I never ever think about it.

This means that, personally, I have to translate "Add child" to "Add property" each time in my head (because that's the language I've always been using for objects).

<img width="718" alt="Screen Shot 2020-01-21 at 14 51 21" src="https://user-images.githubusercontent.com/2854616/72791784-ee8c0400-3c5d-11ea-9122-b4414f2b8092.png">

Reply:
Good points, I hadn't considered that our target user might be engineers (I've been assuming a mix of cluster admins, business analysts, and just general people getting started with the Stack). @jethr0null Could you shed some light on this?

Reply:
I think that at this stage it is better that we stick with the "child/children" naming as it is used in other places (I've seen it reviewing JL PR earlier). Happy to hear  @jethr0null feedback on this though.

Reply:
I chatted with @jethr0null about this and she feels you summed it up perfectly in https://github.com/elastic/kibana/pull/55152#discussion_r368891153. So I leave it up to you! I can go either way way at this point. As Michelle pointed out we can always change the help text later if it ends up confusing folks.